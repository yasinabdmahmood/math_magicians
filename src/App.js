import './App.css';
import React from 'react';
import { Route, Routes } from 'react-router-dom';
import Calculator from './components/Calculator';
import Home from './components/Home';
import Navbar from './components/Navbar';
import Quote from './components/Quote';
// eslint-disable-next-line react/prefer-stateless-function
class App extends React.Component {
  render() {
    return (
      <>
        <Navbar />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/Calculator" element={<Calculator />} />
          <Route path="/Quote" element={<Quote />} />
        </Routes>
      </>

    );
  }
}

export default App;
