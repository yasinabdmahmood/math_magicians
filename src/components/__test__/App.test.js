import { render, screen ,fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from '../../App';
 
const MockApp = () => {
    return (
        <BrowserRouter>
          <App />
        </BrowserRouter>
    )
}

it ("Check if the component has changed", () => {
    const tree = renderer.create(
        <MockApp />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
describe('test the routing functionality',()=>{
    it(' home page should be visible initially ', () => {
        render(
            <MockApp />
        );
        const homePage = screen.getByTestId('home-page');
         
       expect(homePage).toBeVisible();
       
    });

    it('Calculator  page should be visible after clicking on the Calculator link', () => {
        render(
            <MockApp />
        );
        const calculatorLink = screen.getByText("Calculator");
       
        fireEvent.click(calculatorLink)

        const calculatorPage = screen.getByTestId('calculator-page');
         
       expect(calculatorPage).toBeVisible();
       
    });
    it('Quote  page should be visible after clicking on the Quote link', () => {
        render(
            <MockApp />
        );
        const quoteLink = screen.getByText("Quote");
       
        fireEvent.click(quoteLink)

        const quotePage = screen.getByTestId('quote-page');
         
       expect(quotePage).toBeVisible();
       
    });
})
