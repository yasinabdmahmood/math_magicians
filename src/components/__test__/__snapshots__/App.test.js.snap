// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
Array [
  <div
    className="nav-container"
    data-testid="navbar-page"
  >
    <h1>
      Math Magician
    </h1>
    <ul
      className="navbar"
    >
      <li
        data-testid="li"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        data-testid="li"
      >
        <a
          href="/Calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        data-testid="li"
      >
        <a
          href="/Quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="home"
    data-testid="home-page"
  >
    <h1>
      Welcome to our page
    </h1>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint? Sed quibusdam
    </p>
  </div>,
]
`;
